### ISSUE TITLE
context.ShouldRenew does nothing

### ISSUE THREAD

**Issue by pepevc** [aspnet/Security#694]
This is what I have int the Startup class:

```
            app.UseCookieAuthentication(options =>
            {
                options.AuthenticationScheme = "Cookies";
                options.LoginPath = new PathString("/login");
                options.AccessDeniedPath = new PathString("/denied");
                options.AutomaticAuthenticate = true;
                options.ReturnUrlParameter = "url";
                options.Events = new CookieAuthenticationEvents
                {
                    OnValidatePrincipal = async (context) =>
                    {
                        if (context.Principal.Identity != null && context.Principal.Identity.IsAuthenticated)
                        {
                            var userManager = context.HttpContext.RequestServices.GetRequiredService<IUserManager>();

                            if (DateTimeOffset.UtcNow - context.Properties.IssuedUtc > TimeSpan.FromSeconds(int.Parse(Configuration["Authentication:ValidationInterval"])))
                            {
                                User user = await userManager.ValidateSecurityStampAsync(
                                    context.Principal,
                                    context.Principal.Claims.Where(c => c.Type == ClaimTypes.NameIdentifier).SingleOrDefault()?.Value);
                                if (user != null)
                                {
                                    context.ReplacePrincipal(await userManager.CreateClaimsPrincipalAsync(user));
                                    context.ShouldRenew = true;
                                    return;
                                }
                            }
                            return;
                        }
                        context.RejectPrincipal();
                        await context.HttpContext.Authentication.SignOutAsync("Cookies");
                    }
                };
            });
```

The method get calls properly but context.ShouldRenew = true apparently does not make the framework to refresh the cookie.

**Comment by Tratcher** [comment]
What version of Cookies are you using?

@HaoK?

**Comment by pepevc** [comment]
1.0.0-rc1-final

**Comment by Eilon** [comment]
Might be a dup of https://github.com/aspnet/Security/issues/628, which was fixed after RC1.

@HaoK can you confirm?

**Comment by HaoK** [comment]
Yeah this should be fixed already via #628
